resources/js/chat/chat-state-store.ts

```diff
@@ -121,50 +144,59 @@ export default class ChatStateStore implements DispatchListener {
     }
   }
 
-  // TODO: This will support the other types of joining channels in the future
   // Only up to one join/create channel operation should be allowed to be running at any time.
   // For consistency, the operation is considered complete when the websocket message arrives, not when the request completes.
+  // TODO: allow user cancelling operation from UI?
   @action
-  joinChannel() {
-    if (!this.createAnnouncement.isValid || this.isJoiningChannel) return;
-
-    const json = this.createAnnouncement.toJson();
-    this.waitJoinChannelUuid = json.uuid;
-    // TODO: when adding more channel types to join, remember to add separate busy spinner states for them.
-
-    createAnnouncement(json)
-      .fail(action((xhr: JQueryXHR) => {
-        onError(xhr);
-        this.waitJoinChannelUuid = null;
-      }));
+  async joinChannel(channelId?: number) {
```

Comment:
addChannel? Seems weird calling this joinChannel but it sometimes call createAnnouncement instead of (a different) joinChannel

also kinda match the channel id type